src/app/home/home.component.html

```diff
@@ -1,7 +1,12 @@
 <mat-sidenav-container fullscreen>
-  <mat-sidenav [opened]="isSidenavVisible$ | async" mode="side">
-    <cs-sidenav [title]="title" [allowConfiguring]="allowReorderingSidenav$ | async"></cs-sidenav>
+  <mat-sidenav #sidenav>
+    <cs-app-nav [routes]="routes$ | async" [currentRoute]="currentRoute$ | async" (close)="sidenav.close()"></cs-app-nav>
   </mat-sidenav>
 
-  <router-outlet></router-outlet>
+  <mat-sidenav-content>
+    <cs-section-nav [username]="username" [subroutes]="subroutes$ | async" (openAppNav)="sidenav.open()"></cs-section-nav>
+    <div class="primary-content">
+      <router-outlet></router-outlet>
+    </div>
+  </mat-sidenav-content>
 </mat-sidenav-container>
```

Comment:
I have some concerns about these changes:

1. All logic related to exapnding\collapsing of menu items now encapsulated in the template and you can't check it using unit tests.
2. I don't really like the naming of `close` and `openAppNav`. They are using a different naming pattern: one of them includes an action target. It is not so obvious that the events should have specified target, it increases components coupling. For example `closeButtonClicked` sounds better and not overpromising.
3. By these changes, you missed the access to the sidenav state (opened, closed), but it can be helpful in the future.